docs/ref/forms/api.txt

```diff
@@ -830,107 +830,228 @@ The field-specific output honors the form object's ``auto_id`` setting::
     >>> print(f['message'])
     <input type="text" name="message" id="id_message" />
 
-For a field's list of errors, access the field's ``errors`` attribute.
+Attributes of ``BoundField``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. attribute:: BoundField.errors
+.. attribute:: BoundField.auto_id
 
-A list-like object that is displayed as an HTML ``<ul class="errorlist">``
-when printed::
+    Calculates and returns the ID attribute for this BoundField if the
+    associated Form has specified auto_id. Returns an empty string otherwise.
 
-    >>> data = {'subject': 'hi', 'message': '', 'sender': '', 'cc_myself': ''}
-    >>> f = ContactForm(data, auto_id=False)
-    >>> print(f['message'])
-    <input type="text" name="message" />
-    >>> f['message'].errors
-    ['This field is required.']
-    >>> print(f['message'].errors)
-    <ul class="errorlist"><li>This field is required.</li></ul>
-    >>> f['subject'].errors
-    []
-    >>> print(f['subject'].errors)
+    .. note::
+       This attribute gets calculated by the form only, so if the field's
+       widget overrides :meth:`~django.forms.Widget.id_for_label` this ID
+       is likely to be wrong.
```

Comment:
I can think of something like:
```htmldjango
<!-- wrapper div with unique id -->
<div id="{{ field.auto_id }}">
<!-- don't actually use a widget here -->
You can edit "{{ field.label }}" here, but this div
gets replaced dynamically by javascript code.
</div>
```
But I'm not sure if we should encourage writing things like this in a template instead of a custom widget.